decidim-bulletin_board-app/app/commands/close_ballot_box.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# A command with all the business logic to close the election's ballot box
+class CloseBallotBox < Rectify::Command
+  include LogEntryCommand
+
+  # Public: Initializes the command.
+  #
+  # authority - The authority sender of the vote message
+  # message_id - The message identifier
+  # signed_data - The signed message received
+  def initialize(authority, message_id, signed_data)
+    @client = @authority = authority
+    @message_id = message_id
+    @signed_data = signed_data
+  end
+
+  # Executes the command. Broadcasts these events:
+  #
+  # - :ok when everything is valid.
+  # - :invalid if the received data wasn't valid.
+  #
+  # Returns nothing.
+  def call
+    return broadcast(:invalid, error) unless
+      valid_log_entry?("close_ballot_box")
+
+    election.with_lock do
+      return broadcast(:invalid, error) unless
+        valid_step?(election.vote?) &&
```

Comment:
This reminds me that, at the moment, we are not adding date validations on this endpoint. We should have this in mind in the future, to double check that is the desired behavior.

Reply:
Oh yeah, in previous iterations of the _open ballot box_ it was checked in the process message.
I could check here that the date/time is valid.